bitcoin/src/consensus/serde.rs

```diff
@@ -159,6 +159,7 @@ struct DisplayWrapper<'a, T: 'a + Encodable, E>(&'a T, PhantomData<E>);
 impl<'a, T: 'a + Encodable, E: ByteEncoder> fmt::Display for DisplayWrapper<'a, T, E> {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
         let mut writer = IoWrapper::<'_, _, E::Encoder>::new(f, E::default().into());
+        #[allow(unused_variables)] // When debug_assertions are not enabled.
         self.0.consensus_encode(&mut writer).map_err(|error| {
             #[cfg(debug_assertions)]
```

Comment:
I'd prefer `#[cfg(not(debug_assertions))] { let _ = error; }`

Reply:
Thanks, added to issue #3332